src/trio/_tests/check_type_completeness.py

```diff
@@ -0,0 +1,221 @@
+    """Pyright gives us an object identifier of xx.yy.zz
+    This function tries to decompose that into its constituent parts, such that we
+    can resolve it, in order to check whether it has a `__doc__` at runtime and
+    verifytypes misses it because we're doing overly fancy stuff.
+    """
+    assert trio.testing
+
+    # figure out what part of the name is the module, so we can "import" it
+    name_parts = name.split(".")
+    split_i = 1
+    if name_parts[1] == "tests":
+        return True
+    if name_parts[1] in ("_core", "testing"):  # noqa: SIM108
+        split_i = 2
+    else:
+        split_i = 1
+    module = sys.modules[".".join(name_parts[:split_i])]
+
+    # traverse down the remaining identifiers with getattr
+    obj = module
+    try:
+        for obj_name in name_parts[split_i:]:
+            obj = getattr(obj, obj_name)
+    except AttributeError as exc:
+        # asynciowrapper does funky getattr stuff
+        if "AsyncIOWrapper" in str(exc) or name in (
+            # Symbols not existing on all platforms, so we can't dynamically inspect them.
+            # Manually confirmed to have docstrings but pyright doesn't see them due to
+            # export shenanigans. TODO: actually manually confirm that.
+            # darwin
+            "trio.lowlevel.current_kqueue",
+            "trio.lowlevel.monitor_kevent",
+            "trio.lowlevel.wait_kevent",
+            "trio._core._io_kqueue._KqueueStatistics",
+            # windows
+            "trio._socket.SocketType.share",
+            "trio._core._io_windows._WindowsStatistics",
+            "trio._core._windows_cffi.Handle",
+            "trio.lowlevel.current_iocp",
+            "trio.lowlevel.monitor_completion_key",
+            "trio.lowlevel.readinto_overlapped",
+            "trio.lowlevel.register_with_iocp",
+            "trio.lowlevel.wait_overlapped",
+            "trio.lowlevel.write_overlapped",
+            "trio.lowlevel.WaitForSingleObject",
+            "trio.socket.fromshare",
+            # TODO: these are erroring on all platforms, why?
+            "trio._highlevel_generic.StapledStream.send_stream",
+            "trio._highlevel_generic.StapledStream.receive_stream",
+            "trio._ssl.SSLStream.transport_stream",
+            "trio._file_io._HasFileNo",
+            "trio._file_io._HasFileNo.fileno",
+        ):
+            return True
+
+        else:
+            print(
+                f"Pyright sees {name} at runtime, but unable to getattr({obj.__name__}, {obj_name})."
+            )
+            return False
+    return bool(obj.__doc__)
+
+
+def check_type(
+    platform: str, full_diagnostics_file: Path | None, expected_errors: list[object]
+) -> list[object]:
+    # convince isort we use the trio import
+    assert trio
+
+    # run pyright, load output into json
+    res = run_pyright(platform)
+    current_result = json.loads(res.stdout)
+
+    if res.stderr:
+        print(res.stderr)
```

Comment:
This also should do `file=sys.stderr` (won't comment this on future `print`s that are for errors)